### PR TITLE
Implement Internal Docs Collection

### LINKED ISSUE #55: Internal Usage Documentation
Before the school year starts, we should have rigorous documentation that explains how contributors can easily add new information to the website.

Setup work:

- [x] custom docs layout
- [ ] custom docs header
- [ ] custom docs pre-footer
- [x] `_docs` collection and associated rules

Categories that need to be covered:

- [ ] dev setup & git flow
- [ ] adding a new generic page
- [ ] adding new team members (board, alumni, instructors, devs, etc.)
- [ ] adding new classes to the classes page
- [ ] adding new class resources (collection-based)
- [ ] adding new dev projects
- [ ] adding new blog posts
- [ ] adding new reports

Changes that should be either covered or removed from the site:

- [ ] adding new resources
- [ ] adding new about page information

### PR DESCRIPTION
**Pull request by malsf21**
Draft, WIP, the whole nine yards. See #55 

Some notes:

* front page is far from done - should have more CTAs and borders
* i want to make a custom `docs` stylesheet that only includes docs-related styles; avoids collision, and lets us load markdown-generated syntax highlighting easily
* custom nav + better structure?
* thoughts on how word-y the walkthrough is

Docs: https://deploy-preview-68--acm-teach-la-website.netlify.app/docs/

Should close #55

### CODE REVIEW

**Comment by krashanoff** [comment]
Just for a quick update on this:
* I am finishing the internal docs for setting up the dev environment, making a PR, etc. ([dev-setup.md](https://github.com/uclaacm/teach-la-website/blob/implement-docs-collection/_docs/dev-setup.md))
* Will update code highlighting and style. Will likely mirror GitHub's, but try and put a Teach LA twist in there.
* Starting off, we likely won't need a custom nav, but a dropdown nav based on the ToC might be nice for longer docs.

**Review by krashanoff — APPROVED** [review]
Feels pretty weird being the one reviewing a pull request that I contributed to, but here's my two cents:
* Things look good doc-side.
* Changes to the image styling are a little dicey, but they don't affect anything outside the docs negatively right now.
* Once we add more content (which we *will* have a lot of), a custom nav and landing page will be in order.
* Syntax highlighting is a little :thinking:; I think I'll experiment with a few different schemes or borrow the highlighting from our editor and just go with that.

To summarize: I'll say this is good to merge for our purposes and for sake of time. After we add more content, though, another PR will be in order to rework the landing page and the navbar.
